### PR TITLE
player input + movement complete

### PR DESCRIPTION
**Pull request by Raaee**
check it sil vous plait :))))

completed movement and added inputs

### CODE REVIEW

File: Assets/Art/UI/cameraBorder.png
Comment (pistolPeterson) [inline Assets/Art/UI/cameraBorder.png:1]: camera border. sigh

File: Assets/Scripts/Player Input/PlayerMovement.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+{
+    /**
+     * :)
+     * UP -> W / Up Arrow
+     * DOWN -> S / Down Arrow
+     * Click or Hold
+     */
+    [SerializeField] private InputManager inputManager;
+    [SerializeField] private float moveDistance = 1f;
+    [SerializeField] private float upperPositionConstraint = 8f;
```
Comment (pistolPeterson) [inline Assets/Scripts/Player Input/PlayerMovement.cs:15]: the constraints would never be unbalanced, make it one variable please

File: Assets/Scripts/Player Input/InputManager.cs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    }
+    public void ReleasingMove(InputAction.CallbackContext context) {
+        moveHeld = false;
+    }
+    public void UseShield(InputAction.CallbackContext context) {
+        Debug.Log("Used Shield");
+        OnShieldUse.Invoke();
+    }
+    // to allow click and hold movement
+    private IEnumerator Moving() {
```
Comment (pistolPeterson) [inline Assets/Scripts/Player Input/InputManager.cs:62]: wait we have to write our own code to allow multiple button press??????

File: Assets/Scripts/Player Input/PlayerMovement.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    [SerializeField] private float lowerPositionConstraint = -8f;
+
+    private Vector2 moveInput;
+    private void Start() 
+    {inputManager.OnMovement.AddListener(MovePlayer);}
+    public void MovePlayer(Vector2 inputDirection) 
+    {moveInput = inputDirection;  
+    if (moveInput == Vector2.up) 
+    {MovePlayerUp(moveDistance);}
```
Comment (pistolPeterson) [inline Assets/Scripts/Player Input/PlayerMovement.cs:24]: as discussed, make the movement smooth. make the default speed slower than usual so the blessings can make it faster in the future

File: Assets/Scripts/Player Input/InputManager.cs [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        movement.performed -= Move;
+    }
+    public void DisableControls() {
+        movement.Disable();
+        shield.Disable();
+    }
+    public void Move(InputAction.CallbackContext context) {
+        if (alreadyHeld) return;
+        currMoveDir = movement.ReadValue<Vector2>();
+        StartCoroutine(Moving());
```
Comment (pistolPeterson) [inline Assets/Scripts/Player Input/InputManager.cs:49]: this is input system, it should not be named **moving**

**Review by pistolPeterson — CHANGES_REQUESTED** [review]
- smooth movement 
- add a simple visual representation of the shield, maybe the player visual changes color 
- simplify/cleanup input system

File: Assets/Scripts/Player Input/PlayerMovement.cs (hunk 1 above)
Comment (Raaee) [reply Assets/Scripts/Player Input/PlayerMovement.cs]: done.

**Review by pistolPeterson — CHANGES_REQUESTED** [review]
unity nodemap scene doesnt work. unplayable 2/10
code does look good i guess, sigh

**Review by pistolPeterson — APPROVED** [review]
works great, good to go

**Pull request merged**